client/blocks/credit-card-form/index.jsx

```diff
@@ -26,48 +26,54 @@ import support from 'lib/url/support';
 
 const wpcom = wpcomFactory.undocumented();
 
-const CreditCardForm = createReactClass( {
-	displayName: 'CreditCardForm',
+class CreditCardForm extends Component {
+	static displayName = 'CreditCardForm';
 
-	propTypes: {
+	static propTypes = {
 		apiParams: PropTypes.object,
 		createCardToken: PropTypes.func.isRequired,
 		initialValues: PropTypes.object,
 		recordFormSubmitEvent: PropTypes.func.isRequired,
 		saveStoredCard: PropTypes.func,
 		successCallback: PropTypes.func.isRequired,
 		showUsedForExistingPurchasesInfo: PropTypes.bool,
-	},
+	};
 
-	getInitialState() {
-		return {
+	static defaultProps = {
+		apiParams: {},
+		initialValues: {},
+		saveStoredCard: noop,
+		showUsedForExistingPurchasesInfo: false,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
 			form: null,
 			formSubmitting: false,
 			notice: null,
 		};
-	},
-
-	_mounted: false,
-	fieldNames: [
-		'name',
-		'number',
-		'cvv',
-		'expirationDate',
-		'country',
-		'postalCode',
-		'streetNumber',
-		'address1',
-		'address2',
-		'phoneNumber',
-		'streetNumber',
-		'city',
-		'state',
-		'document',
-	],
 
-	componentWillMount() {
-		this._mounted = true;
+		this.fieldNames = [
+			'name',
+			'number',
+			'cvv',
+			'expirationDate',
+			'country',
+			'postalCode',
+			'streetNumber',
+			'address1',
+			'address2',
+			'phoneNumber',
+			'streetNumber',
+			'city',
+			'state',
+			'document',
+		];
+	}
```

Comment:
it should be possible to skip the constructor since our initial state is independent of `props`

```js
static defaultProps = { … }

state = {
	form: null,
	formSubmitting: false,
	notice: null,
}

fieldNames = [
	'name',
	…,
	'document',
]
```